connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -175,6 +175,27 @@ class KustoClient(val clusterAlias: String, val engineKcsb: ConnectionStringBuil
     }
   }
 
+  private[kusto] def setMappingOnStagingTableIfNeeded(ingestionProperties: IngestionProperties, originalTable: String): Unit = {
+    val mapping = ingestionProperties.getIngestionMapping
+    val mappingReferenceName = mapping.getIngestionMappingReference
+    if (StringUtils.isNotBlank(mappingReferenceName)) {
+      val mappingKind = mapping.getIngestionMappingKind.toString
+      val cmd = generateShowTableMappingsCommand(originalTable, mappingKind)
+      val policies = engineClient.execute(ingestionProperties.getDatabaseName, cmd).getValues
+      val it = policies.iterator()
+      var found = false
+      while (it.hasNext && !found){
+        val policy = it.next()
+        if(policy.get(0).equals(mappingReferenceName)){
+          val policyJson = policy.get(2).replace("\"","'")
+          val c = generateCreateTableMappingCommand(ingestionProperties.getTableName, mappingKind, mappingReferenceName, policyJson)
```

Comment:
not sure I follow, what does `ingestionProperties.getTableName` return? the temp table name or the target table name?

Reply:
the ingestion is to the tmpTableName - therefore ingestionProperties.getTableName returns the tmpTable

Reply:
something about the name is inconsistent. 
function name is `setMappingOnStagingTableIfNeeded`, so i would expect the param name to be `stagingIngestionProperites`, and that the calling code will have some variable named accordingly.